### DISCUSSION TITLE
Expanding the Editor based on a Path

### DISCUSSION THREAD

**Discussion by jonathanpbishop** [josdejong/svelte-jsoneditor#448] [Q&A]
Sorry - unfortunately I am confused.  

I want to be able to expand the editor based on path.  I can get it to expend based on the level but not the path. 

I have looked here for examples but I cannot see one.  

I thought I could do something like - 

```

// create a JSONpath variable

      let thisPath = ['processes', 'FIN_1'];

      refEditor.current.expand((path) => path) === thisPath)
```

**Comment by josdejong** [comment]
The function `expand` is powerful but indeed not that easy to understand. Suggestions for a good API are welcome, see #249.

In JavaScript, you can't compare the contents of arrays using `===`, instead you can for example use the [`isEqual`](https://lodash.com/docs/4.17.15#isEqual) function of Lodash. You'll need to check for every sub-path too, like:

```js
import { isEqual } from 'lodash-es'

refEditor.current.expand((path) => {
  return isEqual(path, []) ||
    isEqual(path, ['processes']) ||
    isEqual(path, ['processes', 'FIN_1'])
})
```

Now of course you'll need to write some logic to automatically expand each of the sub-paths.

**Comment by jonathanpbishop** [comment]
Sorry -

Should have replied - that worked great.

And did just what i wanted.

Thanks

Jonathan




Jonathan Bishop

MA PGDip



Principal Process Engineer



D +44 (0)1223 463947             T +44 (0)1223 463500             F +44 (0)1223 461007

***@***.******@***.***>







[cid:65e8e3b4-a3a7-4825-baf2-fe756ee65aa9]

Mott MacDonald
22 Station Road
Cambridge CB1 2JD
United Kingdom





Website<http://www.mottmac.com/>  |  Twitter<https://twitter.com/mottmacdonald>  |  LinkedIn<https://www.linkedin.com/company/7253>  |  Facebook<https://www.facebook.com/mottmacdonaldgroup>  |  YouTube<https://www.youtube.com/user/mottmacdonaldgroup>





Mott MacDonald Limited. Registered in England and Wales no. 1243967. Registered office: Mott MacDonald House, 8-10 Sydenham Road, Croydon CR0 2EE, United Kingdom

The information contained in this e-mail is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. If you are not the intended recipient of this e-mail, the use of this information or any disclosure, copying or distribution is prohibited and may be unlawful. If you received this in error, please contact the sender and delete the material from any computer.


Mott MacDonald Restricted

________________________________

**Reply by josdejong** [reply]
Thanks for letting me know 👍
